tests/v2/test_utils_keyfile.nim

```diff
@@ -0,0 +1,379 @@
+                  "iv": "7e7b02d2b4ef45d6c98cb885e75f48d5",
+              },
+              "ciphertext": "a7a5743a6c7eb3fa52396bd3fd94043b79075aac3ccbae8e62d3af94db00397c",
+              "kdf": "scrypt",
+              "kdfparams": {
+                  "dklen": 32,
+                  "n": 8192,
+                  "p": 1,
+                  "r": 8,
+                  "salt": "247797c7a357b707a3bdbfaa55f4c553756bca09fec20ddc938e7636d21e4a20",
+              },
+              "mac": "5a3ba5bebfda2c384586eda5fcda9c8397d37c9b0cc347fea86525cf2ea3a468",
+          },
+          "address": "0b6f2de3dee015a95d3330dcb7baf8e08aa0112d",
+          "id": "3c8efdd6-d538-47ec-b241-36783d3418b9",
+          "version": 3
+      },
+      "password": "moomoocow",
+      "priv": "21eac69b9a52f466bfe9047f0f21c9caf3a5cdaadf84e2750a9b3265d450d481",
+      "name": "eth-keyfile-conftest"
+    }
+  ]
+
+  test "KeyStoreTests/basic_tests.json test1":
+    var expectkey = decodeHex(TestVectors[0].getOrDefault("priv").getStr())
+    let seckey =
+      decodeKeyFileJson(TestVectors[0].getOrDefault("keyfile"),
+                        TestVectors[0].getOrDefault("password").getStr())
+    check:
+      seckey.isOk()
+      seckey.get() == expectkey
+
+  test "KeyStoreTests/basic_tests.json python_generated_test_with_odd_iv":
+    var expectkey = decodeHex(TestVectors[1].getOrDefault("priv").getStr())
+    let seckey =
+      decodeKeyFileJson(TestVectors[1].getOrDefault("keyfile"),
+                        TestVectors[1].getOrDefault("password").getStr())
+    check:
+      seckey.isOk()
+      seckey.get() == expectkey
+
+  test "KeyStoreTests/basic_tests.json evilnonce":
+    var expectkey = decodeHex(TestVectors[2].getOrDefault("priv").getStr())
+    let seckey = decodeKeyFileJson(TestVectors[2].getOrDefault("keyfile"),
+                      TestVectors[2].getOrDefault("password").getStr())
+    check:
+      seckey.isOk()
+      seckey.get() == expectkey
+  
+  test "KeyStoreTests/basic_tests.json evilnonce with wrong password":
+    let seckey =
+      decodeKeyFileJson(TestVectors[2].getOrDefault("keyfile"),
+                        "wrongpassword")
+    check:
+      seckey.isErr()
+      seckey.error == KeyFileError.IncorrectMac
+
+  test "KeyStoreTests/basic_tests.json test2":
+    var expectkey = decodeHex(TestVectors[3].getOrDefault("priv").getStr())
+    let seckey =
+      decodeKeyFileJson(TestVectors[3].getOrDefault("keyfile"),
+                        TestVectors[3].getOrDefault("password").getStr())
+    check:
+      seckey.isOk()
+      seckey.get() == expectkey
+
+  test "KeyStoreTests/basic_tests.json mycrypto":
+    var expectkey = decodeHex(TestVectors[4].getOrDefault("priv").getStr())
+    let seckey =
+      decodeKeyFileJson(TestVectors[4].getOrDefault("keyfile"),
+                        TestVectors[4].getOrDefault("password").getStr())
+    check:
+      seckey.isOk()
+      seckey.get() == expectkey
+
+  test "eth-key/conftest.py":
```

Comment:
perhaps being more descriptive with the test? something more meaningul and then a link to i.e. https://github.com/ethereum/eth-keys/blob/master/tests/backends/conftest.py? which i belive is the source?